Lib/idlelib/autocomplete.py

```diff
@@ -90,6 +92,126 @@ def _delayed_open_completions(self, args):
         if self.text.index("insert") == self._delayed_completion_index:
             self.open_completions(args)
 
+    _dict_str_key_prefix_re = re.compile(r'\s*(br|rb|fr|rf|b|f|r|u|)', re.I)
+
+    @classmethod
+    def is_completing_dict_key(cls, hp, curline):
```

Comment:
Another minor suggestion. While reading over the changes, the purpose of the *hp* argument in `is_completing_dict_key()` was not clear until I saw line 228. The variable name `hp` isn't as much of an issue on 228 where it's defined, but I think it would make the purpose of the argument in `is_completing_dict_key()` much more clear if it was instead *parser* or *hparser*:

```suggestion
    def is_completing_dict_key(cls, hparser, curline):
```
With this change, anyone reading over the function definition could much more easily understand the purpose of the argument without having to understand the exact implementation details.

Reply:
A fine suggestion, I've just done this.